iso15118/shared/messages/iso15118_2/body.py

```diff
@@ -268,9 +268,11 @@ def requested_energy_mode_must_match_charge_parameter(cls, values):
         if ("AC_" in requested_energy_mode and dc_params) or (
             "DC_" in requested_energy_mode and ac_params
         ):
-            raise ValueError(
+            raise V2GMessageValidationError(
                 "Wrong charge parameters for requested energy "
-                f"transfer mode {requested_energy_mode}"
+                f"transfer mode {requested_energy_mode}",
+                ResponseCode.FAILED_WRONG_CHARGE_PARAMETER,
+                None,
```

Comment:
to make it complete I guess we also need to check that the `requested_energy_mode´ is part of the `EnergyTransferModeEnum` values, otherwise we shall raise a `FAILED_WRONG_ENERGY_TRANSFER_MODE`

Reply:
done